### PR TITLE
Implement MVVM for the Rendering page

### PR DESCRIPTION
**Pull request by PankajBhojwani**
## Summary of the Pull Request
Implements a `RenderingViewModel` for the Rendering page in the SUI

## PR Checklist
* [x] CLA signed. If not, go over [here](https://cla.opensource.microsoft.com/microsoft/Terminal) and sign the CLA
* [x] I've discussed this with core contributors already. If not checked, I'm ready to accept this work might be rejected in favor of a different grand plan. Issue number where discussion took place: #xxx

## Validation Steps Performed
Rendering page still works

### CODE REVIEW

**Review by lhecker — APPROVED** [review]

File: src/cascadia/TerminalSettingsEditor/RenderingViewModel.cpp [hunk 1]
```diff
@@ -0,0 +1,24 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+#include "pch.h"
+#include "RenderingViewModel.h"
+#include "RenderingViewModel.g.cpp"
+#include "EnumEntry.h"
```
Comment (carlos-zamora) [inline src/cascadia/TerminalSettingsEditor/RenderingViewModel.cpp:7]: nit: Looks like we don't need enum entry here?

File: src/cascadia/TerminalSettingsEditor/RenderingViewModel.cpp [hunk 2]
```diff
@@ -0,0 +1,24 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+#include "pch.h"
+#include "RenderingViewModel.h"
+#include "RenderingViewModel.g.cpp"
+#include "EnumEntry.h"
+
+using namespace winrt::Windows::UI::Xaml::Navigation;
```
Comment (carlos-zamora) [inline src/cascadia/TerminalSettingsEditor/RenderingViewModel.cpp:9]: nit: I think we can remove this one

File: src/cascadia/TerminalSettingsEditor/RenderingViewModel.h [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+namespace winrt::Microsoft::Terminal::Settings::Editor::implementation
+{
+    struct RenderingViewModel : RenderingViewModelT<RenderingViewModel>, ViewModelHelper<RenderingViewModel>
+    {
+    public:
+        RenderingViewModel(Model::GlobalAppSettings globalSettings);
+        Model::GlobalAppSettings GlobalSettings();
+
+        WINRT_CALLBACK(PropertyChanged, Windows::UI::Xaml::Data::PropertyChangedEventHandler);
```
Comment (carlos-zamora) [inline src/cascadia/TerminalSettingsEditor/RenderingViewModel.h:18]: We don't use this anywhere?

File: src/cascadia/TerminalSettingsEditor/RenderingViewModel.h [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+#include "ViewModelHelpers.h"
+#include "Utils.h"
+
+namespace winrt::Microsoft::Terminal::Settings::Editor::implementation
+{
+    struct RenderingViewModel : RenderingViewModelT<RenderingViewModel>, ViewModelHelper<RenderingViewModel>
+    {
+    public:
+        RenderingViewModel(Model::GlobalAppSettings globalSettings);
+        Model::GlobalAppSettings GlobalSettings();
```
Comment (carlos-zamora) [inline src/cascadia/TerminalSettingsEditor/RenderingViewModel.h:16]:
```suggestion
        Model::GlobalAppSettings GlobalSettings() const;
```
nit: I think you can add a const here?

**Review by carlos-zamora — COMMENTED** [review]
Can you clean up the PR description too? Some of the checkbox items are unnecessary so you should be able to delete them from the description.

File: src/cascadia/TerminalSettingsEditor/RenderingViewModel.h (hunk 4 above)
Comment (PankajBhojwani) [reply src/cascadia/TerminalSettingsEditor/RenderingViewModel.h]: Yep, thanks for the catch! (and the others too)

**Review by carlos-zamora — APPROVED** [review]

File: src/cascadia/TerminalSettingsEditor/RenderingViewModel.h (hunk 4 above)
Comment (DHowett) [reply src/cascadia/TerminalSettingsEditor/RenderingViewModel.h]: Doesn't actually matter; the Windows Runtime (and COM) doesn't have a notion of const, so really it's never going to care that you've decorated or not-decorated your members

File: src/cascadia/TerminalSettingsEditor/Rendering.xaml [hunk 5]
```diff
@@ -25,13 +25,13 @@
 
             <!--  Force Full Repaint  -->
             <local:SettingContainer x:Uid="Globals_ForceFullRepaint">
-                <ToggleSwitch IsOn="{x:Bind State.Globals.ForceFullRepaintRendering, Mode=TwoWay}"
+                <ToggleSwitch IsOn="{x:Bind ViewModel.GlobalSettings.ForceFullRepaintRendering, Mode=TwoWay}"
```
[the remark this answers is not in the document]
Comment (PankajBhojwani) [reply src/cascadia/TerminalSettingsEditor/Rendering.xaml]: Fixed!

**Pull request merged**